lib/ansible/plugins/connection/aws_ssm.py

```diff
@@ -0,0 +1,469 @@
+# Based on the ssh connection plugin by Michael DeHaan
+#
+# Copyright: (c) 2018, Pat Sharkey <psharkey@cleo.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+author:
+- Pat Sharkey (@psharkey) <psharkey@cleo.com>
+- HanumanthaRao MVL (@hanumantharaomvl) <hanumanth@flux7.com>
+- Gaurav Ashtikar (@gau1991 )<gaurav.ashtikar@flux7.com>
+connection: aws_ssm
+short_description: execute via AWS Systems Manager
+description:
+- This connection plugin allows ansible to execute tasks on an EC2 instance via the aws ssm CLI.
+version_added: "2.8"
+requirements:
+- The remote EC2 instance must be running the AWS Systems Manager Agent (SSM Agent).
+- The control machine must have the aws session manager plugin installed.
+- The remote EC2 instance must have the curl installed.
+options:
+  instance_id:
+    description: The EC2 instance ID.
+    vars:
+    - name: instance_id
+    - name: aws_instance_id
+  region:
+    description: The region the EC2 instance is located.
+    vars:
+    - name: region
+    - name: aws_region
+    default: 'us-east-1'
+  bucket_name:
+    description: The name of the S3 bucket used for file transfers.
+    vars:
+    - name: bucket_name
+  plugin:
+    description: This defines the location of the session-manager-plugin binary.
+    vars:
+    - name: session_manager_plugin
+    default: '/usr/local/bin/session-manager-plugin'
+  retries:
+    description: Number of attempts to connect.
+    default: 3
+    type: integer
+    vars:
+    - name: ansible_ssm_retries
+  timeout:
+    description: Connection timeout seconds.
+    default: 60
+    type: integer
+    vars:
+    - name: ansible_ssm_timeout
```

Comment:
```suggestion
    - name: ansible_aws_ssm_timeout
```

Reply:
Same as above - corrected in https://github.com/ansible/ansible/pull/49652/commits/9b926a0f886a2de90dded4e8113e622debc38868

Reply:
nope, looks like you missed this one.

Reply:
Hi @jamescassell , Thank you for the review and comments. Have updated in https://github.com/ansible/ansible/pull/49652/commits/27360b5f9fd942c93c70c4cdabf6b98f9032c81d